File: CrackingTheCodeInterview/C#/Chapter2/P2_RemoveDuplicates.cs
using System;
using Chapter2;

class RemoveDuplicates
{

    public static void Main()
    {
        SLinkedList sll = new SLinkedList();
        sll.Add(0);
        sll.Add(5);
        sll.Add(19);
        System.Console.WriteLine(sll.last.value);
        System.Console.WriteLine(sll.Size());
        sll.Add(44);
        sll.Add(190);
        System.Console.WriteLine(sll.last.value);
        sll.Add(5);
        sll.Add(19);
        System.Console.WriteLine(sll.last.value);
        sll.Add(44);
        sll.ShowList();
        System.Console.WriteLine(sll.Size());
        sll.RemoveDuplicates();
        sll.ShowList();
        System.Console.WriteLine(sll.last.value);
        System.Console.WriteLine(sll.Size());

    }
}

File: CrackingTheCodeInterview/Python/Chapter2/p2_remove_duplicates.py
from linked_list import SLL

def removing():
    sll = SLL()
    sll.add(1)
    sll.add(3)
    sll.add(1)
    sll.add(10)
    sll.add(2)
    sll.add(3)
    sll.add(10)
    sll.add(5)

    print(sll.size)
    sll.show_list()
    sll.remove_duplicates()
    print(sll.size)
    sll.show_list()

if __name__ == '__main__':
    removing()
